src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -46,11 +45,25 @@
 public class HighlightPhase extends AbstractComponent implements FetchSubPhase {
 
     private final Highlighters highlighters;
+    private final ImmutableList<Highlighter> sortedHighlighters;
 
     @Inject
     public HighlightPhase(Settings settings, Highlighters highlighters) {
         super(settings);
         this.highlighters = highlighters;
+        // TODO: would be nice if each highlighter would have a precedence and we could just sort here by this value
+        ImmutableList.Builder sortedListBuilder = ImmutableList.builder();
+        sortedListBuilder.add(highlighters.get("fvh"));
+        sortedListBuilder.add(highlighters.get("postings"));
+        sortedListBuilder.add(highlighters.get("plain"));
+        for (Map.Entry highlighter : highlighters.entrySet()) {
```

Comment:
This probably isn't required - other highlighters can just be called out in the request if you want them.